packages/cms-cli/commands/lint.js

```diff
@@ -27,51 +27,75 @@ const {
 } = require('../lib/usageTracking');
 
 const COMMAND_NAME = 'lint';
+const DESCRIPTION = 'Lint a file or folder for HubL syntax';
 
-function configureCommand(command) {
-  command
-    .version(version)
-    .description('lint a file or folder for HubL syntax')
-    .arguments('<path>')
-    .action(async filepath => {
-      setLogLevel(command);
-      logDebugInfo(command);
-      const { config: configPath } = command;
-      loadConfig(configPath);
-      checkAndWarnGitInclusion();
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath);
+  checkAndWarnGitInclusion();
 
-      if (!(validateConfig() && (await validatePortal(command)))) {
-        process.exit(1);
-      }
-
-      const portalId = getPortalId(command);
+  if (!(validateConfig() && (await validatePortal(options)))) {
+    process.exit(1);
+  }
+};
 
-      trackCommandUsage(COMMAND_NAME, {}, portalId);
+const action = async (args, options) => {
+  const portalId = getPortalId(options);
+  const localPath = resolveLocalPath(args.localPath);
+  const groupName = `Linting "${localPath}"`;
 
-      filepath = resolveLocalPath(filepath);
+  trackCommandUsage(COMMAND_NAME, {}, portalId);
+  await loadAndValidateOptions(options);
```

Comment:
I think moving this logic below `getPortalId()` is likely to cause problems because we haven't validated the portal.

Reply:
Ahhh that makes sense.  Thanks for that.  Putting it back at the top